vavr/src/main/java/io/vavr/control/Option.java

```diff
@@ -361,6 +361,18 @@ default T getOrElse(Supplier<? extends T> supplier) {
     }
 
     /**
+     * Folds either the {@code None} or the {@code Some} side of the Option value.
+     *
+     * @param mapperNone  maps the left value if this is a None
+     * @param mapperSome maps the right value if this is a Some
+     * @param <U>         type of the folded value
+     * @return A value of type U
+     */
+    default <U> U fold(Supplier<? extends U> mapperNone, Function<? super T, ? extends U> mapperSome) {
```

Comment:
The supplier does not map an element. Could we use the same parameter names like Scala?

E.g. (modulo generics):

```java
fold(Supplier ifEmpty, Function f)
```

or

```java
fold(Supplier ifEmpty, Function mapper)
```